pkg/core/transactor/transactor.go

```diff
@@ -2,191 +2,115 @@ package transactor
-
-	// Create a new stake tx
-	tx, err := t.w.NewStakeTx(cfg.MinFee, lockTime, amountScalar)
-	if err != nil {
-		return nil, err
+	if err := t.rb.Register(rpcbus.LoadWallet, loadWalletChan); err != nil {
+		return err
 	}
 
-	// Sign tx
-	err = t.w.Sign(tx)
-	if err != nil {
-		return nil, err
+	if err := t.rb.Register(rpcbus.CreateWallet, createWalletChan); err != nil {
+		return err
 	}
 
-	return tx, nil
-}
-
-func (t *Transactor) CreateBidTx(amount, lockTime uint64) (transactions.Transaction, error) {
-	if err := t.syncWallet(); err != nil {
-		return nil, err
+	if err := t.rb.Register(rpcbus.CreateFromSeed, createFromSeedChan); err != nil {
+		return err
 	}
 
-	// Turn amount into a scalar
-	amountScalar := ristretto.Scalar{}
-	amountScalar.SetBigInt(big.NewInt(0).SetUint64(amount))
-
-	// Create a new bid tx
-	tx, err := t.w.NewBidTx(cfg.MinFee, lockTime, amountScalar)
-	if err != nil {
-		return nil, err
+	if err := t.rb.Register(rpcbus.SendBidTx, sendBidTxChan); err != nil {
+		return err
 	}
 
-	// Sign tx
-	err = t.w.Sign(tx)
-	if err != nil {
-		return nil, err
+	if err := t.rb.Register(rpcbus.SendStakeTx, sendStakeTxChan); err != nil {
+		return err
 	}
 
-	return tx, nil
-}
+	if err := t.rb.Register(rpcbus.SendStandardTx, sendStandardTxChan); err != nil {
+		return err
+	}
 
-func (t *Transactor) syncWallet() error {
-	t.syncLock.Lock()
-	defer t.syncLock.Unlock()
-	var totalSpent, totalReceived uint64
-	// keep looping until tipHash = currentBlockHash
-	for {
-		// Get Wallet height
-		walletHeight, err := t.w.GetSavedHeight()
-		if err != nil {
-			t.w.UpdateWalletHeight(0)
-		}
-
-		// Get next block using walletHeight and tipHash of the node
-		blk, tipHash, err := fetchBlockHeightAndState(t.db, walletHeight)
-		if err == database.ErrBlockNotFound {
-			break
-		}
-		if err != nil {
-			return fmt.Errorf("error fetching block from node db: %v\n", err)
-		}
-
-		// call wallet.CheckBlock
-		spentCount, receivedCount, err := t.w.CheckWireBlock(*blk)
-		if err != nil {
-			return fmt.Errorf("error checking block: %v\n", err)
-		}
-
-		totalSpent += spentCount
-		totalReceived += receivedCount
-
-		// check if state is equal to the block that we fetched
-		if bytes.Equal(tipHash, blk.Header.Hash) {
-			break
-		}
+	if err := t.rb.Register(rpcbus.GetBalance, getBalanceChan); err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func fetchBlockHeightAndState(db database.DB, height uint64) (*block.Block, []byte, error) {
-	var blk *block.Block
-	var state *database.State
-	err := db.View(func(t database.Transaction) error {
-		hash, err := t.FetchBlockHashByHeight(height)
-		if err != nil {
-			return err
-		}
-		state, err = t.FetchState()
-		if err != nil {
-			return err
-		}
-
-		blk, err = t.FetchBlock(hash)
-		if err != nil {
-			return err
-		}
-		return nil
-	})
-	if err != nil {
-		return nil, nil, err
-	}
-
-	return blk, state.TipHash, nil
+// Collector implements the wire.EventCollector interface
+type Collector struct {
```

Comment:
We already have an `AcceptedBlockCollector` in the consensus package I believe, so this is redundant.

Reply:
If AcceptedBlockCollector can be used as a common util by different packages, it should be out of consensus pkg. 
I suggest we have another issue to make all pkgs that are collecting acceptedBlock use AcceptedBlockCollector from `utils` pkg.

Reply:
That sounds fine to me, but is there a reason you wrote this extra code? We should probably limit the amount of stuff to fix after this PR is merged, as we already have some skipped tests.